joomla/joomla-cms
# Semantic image captions

## Overview

Following the Google Group discussion regarding jCaption https://groups.google.com/forum/#!topic/joomla-dev-general/vW--mTtuaaw this PR implements semantic figure and captioning elements.

Joomla ships with the html5 doctype, so it is my opinion that we should be using the correct elements wherever possible. Article images and their captions are what  and  are made for.

This change not only improves the document semantics, and therefore accessibility, it also removes the need for caption.js, which calls the entire Mootools stack whenever it is loaded.
## Changes

This PR makes several changes, namely:
1. Disabling calls to jCaption and marking as deprecated.
2. Updating markup surrounding article intro and article full text images to implement html5s figure and figcaption elements.
3. Updates the featured layout to use the JLayout intro image helper, as the category blog layout already does.
## Testing
1. Insert article images (intro and full-text) using the 'images and links' fields.
2. Insert image captions using the 'images and links' Caption field, one for intro image, one for full-text.
3. Set the article to 'featured' status.
4. Check that the image caption appears in article, category blog and featured blog view.
5. Check that the 'Image float' option is honored in these views.
6. Check that caption.js is no longer loaded in these views.
